ui/components/app/import-token/network-selector-custom-import/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  TextVariant,
+  TextColor,
+  BorderRadius,
+  AlignItems,
+  Display,
+  JustifyContent,
+  BorderColor,
+} from '../../../../helpers/constants/design-system';
+import {
+  ButtonIcon,
+  Box,
+  ButtonIconSize,
+  IconName,
+  Text,
+  AvatarNetwork,
+  AvatarNetworkSize,
+} from '../../../component-library';
+import { getNetworkConfigurationsByChainId } from '../../../../../shared/modules/selectors/networks';
+import { getImageForChainId } from '../../../../selectors/multichain';
+
+export const NetworkSelectorCustomImport = ({
+  title,
+  buttonDataTestId,
+  chainId,
+  onSelectNetwork,
+}: {
+  title: string;
+  buttonDataTestId: string;
+  chainId: string;
+  onSelectNetwork: () => void;
+}) => {
+  const allNetworks = useSelector(getNetworkConfigurationsByChainId);
+  const networkImageUrl = getImageForChainId(chainId);
+
+  const allOpts: Record<string, boolean> = {};
+  Object.keys(allNetworks || {}).forEach((chain) => {
```

Comment:
I see this `Object.keys` usage in other files like `ui/components/app/assets/asset-list/asset-list-control-bar/asset-list-control-bar.tsx` and `ui/components/app/assets/asset-list/network-filter/network-filter.tsx`.  Can we put this in a selector and reuse the logic?

Can we put this in a